pkg/skaffold/deploy/kubectl/debug.go

```
@@ -0,0 +1,316 @@
+			}
+		}
+		for i := range allocated {
+			if !allocated[i] {
+				return desiredPort + int32(i)
+			}
+		}
+		// on to the next window
+		desiredPort += windowSize
+	}
+	// NOTREACHED
+}
+
+// imageConfiguration captures information from a docker/oci image configuration
+type imageConfiguration struct {
+	labels     map[string]string
+	env        map[string]string
+	entrypoint []string
+	arguments  []string
+}
+
+func retrieveImageConfiguration(image string, artifact build.Artifact) imageConfiguration {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	config, err := artifact.Config(ctx)
+	if err != nil {
+		logrus.Errorf("unable to retrieve image configuration for %q: %v", image, err)
+		return imageConfiguration{}
+	}
+	return imageConfiguration{
+		env:        envAsMap(config.Env),
+		entrypoint: config.Entrypoint,
+		arguments:  config.Cmd,
+		labels:     config.Labels,
+	}
+}
+
+// transformContainer rewrites the container definition to enable debugging and returns a debugging configuration description
+func transformContainer(container *v1.Container, artifact build.Artifact, portAlloc portAllocator) map[string]interface{} {
+	config := retrieveImageConfiguration(container.Image, artifact)
+
+	// update image configuration values with those set in the k8s manifest
+	for _, envVar := range container.Env {
+		// FIXME handle ValueFrom?
+		config.env[envVar.Name] = envVar.Value
+	}
+
+	if len(container.Command) > 0 {
+		config.entrypoint = container.Command
+	}
+	if len(container.Args) > 0 {
+		config.arguments = container.Args
+	}
+
+	switch guessRuntime(config) {
+	case JVM:
+		logrus.Debugf("Configuring %v for JVM", container.Name)
+		return configureJvmDebugging(container, config, portAlloc)
+	case NODEJS:
+		logrus.Debugf("Configuring %v for NODEJS", container.Name)
+		return configureNodeJSDebugging(container, config, portAlloc)
+	default:
+		logrus.Debugf("Unable to determine runtime for %v\n", container.Name)
+		return nil
```

maybe return an error here so that skaffold debug fails for unsupported runtimes instead of silently passing.